### DISCUSSION TITLE
Different Truncation for Correlated Parameters?

### DISCUSSION THREAD

**Discussion by jjasonbell** [TuringLang/Turing.jl#1813] [Q&A]
The context is that I'm trying to estimate a Multivariate Probit model in Turing. See this Stan page for a writeup: https://mc-stan.org/docs/2_18/stan-users-guide/multivariate-outcomes.html#ref-AlbertChib:1993

Is this possible in Turing?

It seems the key missing piece is the ability to set bounds on parameters separately from the distribution the parameter is drawn from. In Stan when you declare the parameter you can set a bound, and that gets enforced even if the parameter's distribution would normally have support outside of the bound.

I am very new to Turing, and it appears this isn't possible, but I'd love to be able to use Turing for this problem if I can.

**Comment by devmotion** [comment]
Maybe you're looking for https://juliastats.org/Distributions.jl/stable/truncate/ or https://juliastats.org/Distributions.jl/stable/censored/? In Turing you should be able to use any distribution type that follows the Distributions.jl interface (if not, it's a bug).

Alternatively, one can enforce bounds manually by setting the log probability to -Inf for undesired values (can be slow though) or implement a custom distribution.

**Reply by jjasonbell** [reply]
I looked into the TruncatedNormal from Distributions.jl. The problem here is I will need to have a vector, say `z ~ MvNormal(mu, Sigma)`, but some of the elements of `z` need to be negative and others positive, according to values of data `y` of the same shape as `z`.  The `UnivariateDistribution` type here is the hangup:

    truncated(d0::UnivariateDistribution; [lower::Real], [upper::Real])
    truncated(d0::UnivariateDistribution, lower::Real, upper::Real)

A custom distribution might work. I'll look into it. Something like a new distribution type `TruncatedMvNormal(mu, Sigma, y)`
